### DISCUSSION TITLE
og: tag for first image

### DISCUSSION THREAD

**Discussion by DigitalTrading** [openstore-ecommerce/OpenStore#297] [General]
Hi,
I have added a pinterest share button to each product, however it is only pulling the sites logo. Can you advise how I can assign the open graph tag to the first image assigned to the product?

Thanks 
Roger

**Comment by DNNMonster** [comment]
I am not working with that widget so I am not 100% sure about the implementation.  Do you have to have a graph tag in the head?  Or is the a custom link that can be done in the body of the html?   They would both be custom template modifications and I would probably try to use js from the product detail template to update the graph tag in the head. You can set the graph tag in a custom os_FrontOfficePageHeader.cshtml.  Then you can update the image url using js from within a custom ProductDisplayDetail.cshtml template.   You may also be able to accomplish the task by only modifying the ProductDisplayDetail and appending the head with a graph tag by using js.   If someone knows a better way then hopefully they'll speak up.

**Comment by DigitalTrading** [comment]
That was really helpful,
this is what we added to the os_FrontOfficePageHeader.cshtml file
@{
    var productData = new ProductData();
    var Context = HttpContext.Current;
    var entryId = Utils.RequestQueryStringParam(Context, "eid");
    if (Utils.IsNumeric(entryId))
    {
        productData = ProductUtils.GetProductData(Convert.ToInt32(entryId), Utils.GetCurrentCulture(), true, "PRD");
    }
}
@if (productData.Exists)
{
<meta property = "og:title" content="@productData.SEOName" />
<meta property = "og:description" content="@productData.SEODescription" />
<meta property = "og:image" content="@("https://www.printableparties.com" + ProductImageUrl(productData.Info, 0, 0, 1))" />
<meta property = "og:url" content="@("https://www.printableparties.com" + Context.Request.RawUrl.ToString())" />
<meta property="og:type" content="article" />
}
Hope this helps others.

Thanks again
